### PR TITLE
v1.5.0 - MotoROS Driver

### PR DESCRIPTION
**Pull request by ted-miller**
Add support for YRC1000 controller. (Not compatible with Human Collaborative features.)
Add support to control an entire I/O Group (byte).

### CODE REVIEW

File: motoman_driver/MotoPlus/SimpleMessage.h [hunk 1]
```diff
@@ -59,10 +59,14 @@ typedef enum
...
-	ROS_MSG_MOTO_WRITE_SINGLE_IO = 2005,
-	ROS_MSG_MOTO_WRITE_SINGLE_IO_REPLY = 2006,
+	ROS_MSG_MOTO_READ_IO_BIT = 2003,
+	ROS_MSG_MOTO_READ_IO_BIT_REPLY = 2004,
+	ROS_MSG_MOTO_WRITE_IO_BIT = 2005,
+	ROS_MSG_MOTO_WRITE_IO_BIT_REPLY = 2006,
+	ROS_MSG_MOTO_READ_IO_GROUP = 2007,
+	ROS_MSG_MOTO_READ_IO_GROUP_REPLY = 2008,
+	ROS_MSG_MOTO_WRITE_IO_GROUP = 2009,
+	ROS_MSG_MOTO_WRITE_IO_GROUP_REPLY = 2010,
```
Comment (gavanderhoorn) [inline motoman_driver/MotoPlus/SimpleMessage.h:69]: If we want to do this properly, we'd need to update [REP-I0004](https://github.com/ros-industrial/rep/blob/master/rep-I0004.rst), specifically section [Vendor Specific Messages - Motoman](https://github.com/ros-industrial/rep/blob/master/rep-I0004.rst#motoman).

Comment (ted-miller) [reply motoman_driver/MotoPlus/SimpleMessage.h:69]: See [PR18](https://github.com/ros-industrial/rep/pull/18)

**Comment by ted-miller** [comment]
@shaun-edwards  Friendly ping.
I think that Eric's upcoming PR will be easier for us git-newbs (myself and Eric) if this is merged first.

**Review by shaun-edwards — APPROVED** [review]

**Comment by shaun-edwards** [comment]
@ted-miller, do you want this changed ported back to ROS-Indigo?

**Comment by ted-miller** [comment]
That's fine with me. I'm not really sure what versions people are using, but this piece is pretty universal.



Sent from my Verizon, Samsung Galaxy smartphone


-------- Original message --------

**Comment by BlazingForests** [comment]
Hi, guys, Can i use MotoPlus SDK for YRC1000 to compile MotoRos code for DX200 controller?

THX!

**Comment by gavanderhoorn** [comment]
Do you just want to run the current MotoROS code, or do you want to edit it and then compile it? If the former, be aware that binaries for the DX200 can be found [here](https://github.com/ros-industrial/motoman/tree/1289a8149d2debc7e9ed82f9abf8a44022a171b7/motoman_driver/MotoPlus/output/DX200/MotoRosDX200) already.

**Comment by ted-miller** [comment]
As gavanderhoorn mentioned, the binaries are already available on Github and also posted to the wiki.

If you want to edit the MotoROS source code and compile, then you will need the MotoPlus SDK specifically for DX200.  The compiler for each controller model is licensed separately.

**Comment by BlazingForests** [comment]
@ted-miller @gavanderhoorn Thank you! I got it.

So, I have an question. If i have 2 or 3 DX200, Do i need purchase multi MotoPlus SDK for each controller?

**Comment by ted-miller** [comment]
No, you only one SDK for each model/generation (DX200, YRC1000, etc...)

**Comment by BlazingForests** [comment]
@ted-miller Thank you very much!

**Comment by BlazingForests** [comment]
Sorry, One more question, Are parameters setting and setup of YRC1000 for MotoROS the same as DX200? THX!

**Comment by gavanderhoorn** [comment]
@BlazingForests: could I ask you to please ask this question either in a new issue here on the tracker or on ROS Answers? That way it will be more visible to future readers that might have the same question(s).

**Comment by BlazingForests** [comment]
@gavanderhoorn Sure! Sorry for the oversight.

**Pull request merged**